### ISSUE TITLE
Need old version

### ISSUE THREAD

**Issue by Alex313031** [lefterisgar/caprine#14]
Why did you take down the old releases!! I hate when anyone does that, be it github or elsewhere, you cant assume that an older release may never be needed again.

The latest releases from upstream are freezing, only the content, not the actual electron browser window.

v2.55.7-2 worked fine, and I went to the releases here to grab it, but you removed everything but the latest windows release and I am on ubuntu jammy.

**Comment by lefterisgar** [comment]
> Why did you take down the old releases!! I hate when anyone does that, be it github or elsewhere, you cant assume that an older release may never be needed again.
> 
> The latest releases from upstream are freezing, only the content, not the actual electron browser window.
> 
> v2.55.7-2 worked fine, and I went to the releases here to grab it, but you removed everything but the latest windows release and I am on ubuntu jammy.

I took down the releases, but not the source! You can always check out the old tag and build from source if you wish! I'm sorry you have problems but we are trying to fix it. Plus the old releases are broken because Messenger has changed the UI again and broke everything.

**Comment by Alex313031** [comment]
@lefterisgar I think what I might be experiencing is https://github.com/sindresorhus/caprine/issues/1878 I am on ubuntu 22.04, and have used the .deb, as well  as the .appimage, as well as building it myself, and just making a .desktop file that calls npm start lol.

It because anytime I switch to another application, and then go back to caprine, the viewport will freeze. The menubar still works (although clicking on stuff wont do anything, except quit). But you cant reload, and the only thing to do is quit it an manually start it again. IDK if it is an electron issue though, because I have some programs using electron 19, 20. And a project I'm working on is using 21 nightly, built from source. They all seem to work fine on this machine as well as a machine running 18.04.

**Comment by lefterisgar** [comment]
> @lefterisgar I think what I might be experiencing is [sindresorhus#1878](https://github.com/sindresorhus/caprine/issues/1878) I am on ubuntu 22.04, and have used the .deb, as well as the .appimage, as well as building it myself just making a .desktop file that calls npm start lol.
> 
> It's because anytime I switch to another application, and then go back to caprine, the viewport will freeze. The menubar still works (although clicking on stuff wont do anything, except quit). But you cant reload, and the only thing to do is quit it and manually start it again. IDK if it is an electron issue though like some including you have suggested, because I have some programs using electron 19, & 20. And a project I'm working on is using 21 nightly, built from source. They all seem to work fine on this machine as well as a machine running 18.04.

Yes, the linked issue is probably what you are experiencing. It might be that with newer Electron versions something has changed (an API or something else) that breaks some Caprine functionality and causes those issues. That's why some apps don't have it.

**Comment by Alex313031** [comment]
@lefterisgar The 2.55.7-3 is working just fine, no css issues or anything! But how do I package it. All it says on both your repo and upstream is just "see electron builder docs" which isnt really helpful. I cant figure out how to make a package. I would like a .deb and a .zip file. Because right now im just using npm start. Packaging it also minifies stuff and makes it faster, correct?

**Comment by lefterisgar** [comment]
My fork provides instructions on how to package it. You were probably on the official branch and didn't switch it.
Anyways, taken from my fork:

`node_modules/.bin/electron-builder --linux Appimage deb rpm`

This command will create an AppImage and native packages for Debian & Fedora.

> Packaging it also minifies stuff and makes it faster, correct?

Yeah, I assume the same thing!

**Comment by lefterisgar** [comment]
Since you are an Electron expert, could it be that I don't experience any of those issues (lag, freezing, etc.) because on my testing machine I use Node 18? If so I could change that on the official version as well.

**Comment by Alex313031** [comment]
@lefterisgar Im using jammy, which comes with 12.9, but I use n from npm to install the latest stable version in /usr/share/local.

This is 16.17

Perhaps you should try with that.

And I'm not an electron expert, I'm a Chromium expert. Although electron is built on chromium, I am pretty dumb when it comes to actually trying to make anything with electron.

**Comment by lefterisgar** [comment]
> And I'm not an electron expert, I'm a Chromium expert. Although electron is built on chromium, I am pretty dumb when it comes to actually trying to make anything with electron.

I see. For someone who doesn't have a lot of knowledge of the subject, it's easy to confuse the two.

> Perhaps you should try with that.

Thanks! I'm going to experiment a bit and see!

I really appreciate your help and hope that a fix is soon found for this annoying issue. Also, I saw your web browser and I'm pretty amazed. Looking forward to trying it out!

**Comment by Alex313031** [comment]
@lefterisgar Thanks for your work. You allowed me to use caprine when the upstream wouldnt work without manually extracting the .deb and running it with --no-sandbox. I can see he saw your work since you are now part of the team.

And yeah, thorium is my main project, which is why it is my PFP. 

I make builds for linux, windows, macos, and raspberry pi..

Heres the site for it > https://thorium.rocks/ as well as two new pages I recently made.

First, a write up about the optimizations it uses > https://thorium.rocks/optimizations.html

Second, a page comparing performance with screenshots against vanilla chromium, chrome, brave, and vivaldi. > https://thorium.rocks/performance
